hazelcast/src/hazelcast/client/xml_impl.cpp

```diff
@@ -0,0 +1,1317 @@
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+#include <boost/algorithm/string/case_conv.hpp>
+#include <boost/algorithm/string/trim.hpp>
+#include <boost/algorithm/string/join.hpp>
+#include "hazelcast/client/client_config.h"
+#include "hazelcast/client/serialization_config.h"
+#include "hazelcast/client/config/ssl_config.h"
+#include "hazelcast/client/config/client_flake_id_generator_config.h"
+#include "hazelcast/client/exception/protocol_exceptions.h"
+#include "hazelcast/client/internal/partition/strategy/StringPartitioningStrategy.h"
+#include "hazelcast/client/address.h"
+#include "hazelcast/client/config/client_network_config.h"
+#include "hazelcast/client/config/client_aws_config.h"
+#include "hazelcast/client/config/reliable_topic_config.h"
+#include "hazelcast/client/config/client_connection_strategy_config.h"
+#include "hazelcast/client/lifecycle_listener.h"
+#include "hazelcast/client/initial_membership_event.h"
+#include "hazelcast/client/internal/config/xml_config_locator.h"
+#include "hazelcast/client/internal/config/xml_dom_config_processor.h"
+#include "hazelcast/client/internal/config/xml_variable_replacer.h"
+#include "hazelcast/client/internal/config/xml_config_builder.h"
+#include "hazelcast/client/hazelcast_client.h"
+#include <utility>
+#include <boost/property_tree/xml_parser.hpp>
+
+
+namespace hazelcast {
+namespace client {
+namespace internal {
+namespace config {
+
+abstract_dom_config_processor::abstract_dom_config_processor(bool dom_level_3)
+{
+    this->dom_level_3 = dom_level_3;
+    this->strict = true;
+}
+abstract_dom_config_processor::abstract_dom_config_processor(bool dom_level_3,
+                                                             bool strict)
+{
+    this->dom_level_3 = dom_level_3;
+    this->strict = strict;
+}
+
+bool
+abstract_dom_config_processor::matches(const std::string& config1,
+                                       const std::string& config2)
+{
+    return config1 == config2;
+}
+std::string
+abstract_dom_config_processor::get_attribute(boost::property_tree::ptree node,
+                                             const std::string& attribute)
+{
+    return node.get_child("<xmlattr>." + attribute).data();
+}
+bool
+abstract_dom_config_processor::get_bool_value(std::string value)
+{
+    boost::algorithm::to_lower(value);
+    if (value == "true") {
+        return true;
+    } else if (value == "false") {
+        return false;
+    }
+    return false;
```

Comment:
It can be simplified by :

```suggestion
    return boost::algorithm::to_lower(value) == "true";
```